evalml/pipelines/components/transformers/imputers/simple_imputer.py

```diff
@@ -86,17 +103,13 @@ def transform(self, X, y=None):
         not_all_null_cols = [col for col in X.columns if col not in self._all_null_cols]
         original_index = X.index
 
-        # Not using select because we just need column names, not a new dataframe
-        X.ww.set_types(
-            {
-                col: "Categorical"
-                for col, ltype in X.ww.logical_types.items()
-                if isinstance(ltype, NaturalLanguage)
-            }
-        )
+        X_t, natural_language_columns = self._drop_natural_language_columns(X)
```

Comment:
`natural_language_columns` is unused. Can we drop it from the return type?

Reply:
Donezo